### ISSUE TITLE
Allow github_remotes() to search for non-standard github configs

### ISSUE THREAD

**Issue by BarkleyBG** [r-lib/usethis#797]
I am using two git profiles on my computer. The one I want to setup with a new package and usethis has the following information in the `/Users/me/.ssh/config` file:

```sh
Host my_other_account
 HostName github.com
 User git
 IdentityFile ~/.ssh/my_other_rsa
```

Since the host name is `my_other_account`, then `usethis:::github_remotes()` does not recognize it as a github remote:
https://github.com/r-lib/usethis/blob/2a862fa9c3420ff9282896331053c9fbaf12a54a/R/github-utils.R#L1-L4
That is, line 2 returns the following, as is intended:
```r
$origin
[1] "my_other_account:BarkleyBG/my_repo.git"
```
however, line 4 greps for the phrase `"github"` and fails, even though the git alias is setup correctly.

Major takeaways:
1. Did I miss an existing feature of `usethis` or `remotes` that would help remedy this?
2. Is there a way to programmatically fix this?
3. I did come up with a solution: I changed my `~/.ssh/config` file so that it reads the following way, and I use the second config alias with this repo and it works fine with usethis (so far):

```sh
Host my_other_account
 HostName github.com
 User git
 IdentityFile ~/.ssh/my_other_rsa

Host my_other_github_account
 HostName github.com
 User git
 IdentityFile ~/.ssh/my_other_rsa
```

Thank you for your help! Especially Jenny's in this kind twitter reply: https://twitter.com/JennyBryan/status/1116097484248371201

**Comment by jennybc** [comment]
> Did I miss an existing feature of usethis or remotes that would help remedy this?

No.

> Is there a way to programmatically fix this?

No.

> I did come up with a solution: I changed my ~/.ssh/config file so that it reads the following way, and I use the second config alias with this repo and it works fine with usethis (so far) ...

There's currently no direct connection between usethis and `~/.ssh/config`. As you've found, we just do a rather naive analysis of the remote URLs.

I don't have any personal experience with what you're doing re: multiple git profiles, but I have a vague sense of what you're talking about. Can you point me towards a good resource on this?

**Comment by BarkleyBG** [comment]
I think this is one of the better resources I've found (the accepted answer to this S/O question): https://stackoverflow.com/questions/3860112/multiple-github-accounts-on-the-same-computer

I said "I did come up with a solution" but I think I should have said "_workaround_" ! 

In the examples of the `~/.ssh/config` files above, it seems that `Host` acts like a nickname that can be specified in various manners, but I think that `HostName` refers to the site that the ssh keypair is sent to for validation anytime an ssh command is sent. 

- I suppose a programmatic solution would be, for any `Host` provided, to try to see if the `HostName` was github.com. 
- To do this: 
  - I imagine that scraping the the `~/.ssh/config` would not be easy, 
  - I imagine that someone could figure out how R could ask git to return the `HostName`, but it's beyond me how that could work at moment.

TLDR: Perhaps the best way forward is to note that a good workaround is to use the phrase "github" in the `Host` for any non-standard git profile / ssh keypair.

**Comment by BarkleyBG** [comment]
Here's two more references that I've used in the past:

- an S/O response to [setup two ssh-keypairs and configure a different psuedo-host for each account](https://stackoverflow.com/a/9348040)
- a GH Gist on [Multiple SSH Keys settings for different github account](https://gist.github.com/jexchan/2351996)

**Comment by hadley** [comment]
I think this unfortunately this is a sufficiently rare case that the best option is to just rely on people finding this issue when googling.
